### DISCUSSION TITLE
404 error during octokit pull request

### DISCUSSION THREAD

**Discussion by abhimehta23** [octokit/octokit.js#2239] [Q&A] [answered]
Hi @gr2m ,

I am using Octokit npm package to retrieve list of pull requests from an Enterprise Github account by passing my personal access token as an auth strategy. For some reason I have been getting 404 message in the status after running `await octokit.request('GET /{owner}/{repo}/pulls')`

I have confirmed the url returned by the method is also correct and the token setting also include "repo" as enabled but still not luck. Please see message below:

```
status: 404,
  response: {
    url: 'https://github.com/{owner}/{repo}/pulls?branch=abhi',
    status: 404,
    headers: {
      'cache-control': 'no-cache',
      connection: 'close',
      'content-encoding': 'gzip',
      'content-security-policy': "default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'",
      'content-type': 'text/plain; charset=utf-8',
      date: 'Thu, 30 Jun 2022 22:24:35 GMT',
      'expect-ct': 'max-age=2592000, report-uri="https://api.github.com/_private/browser/errors"',
      'permissions-policy': 'interest-cohort=()',
      'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
      server: 'GitHub.com',
      'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
      'transfer-encoding': 'chunked',
      vary: 'X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Encoding, Accept, X-Requested-With',
      'x-content-type-options': 'nosniff',
      'x-frame-options': 'deny',
      'x-github-request-id': 'F077:3789:86FA73:A9F0DA:62BE22A3',
      'x-xss-protection': '0'
    },
    data: 'Not Found'
  },
  request: {
    method: 'GET',
    url: 'https://github.com/{owner}/{repo}/pulls?branch=abhi',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit.js/1.8.1 octokit-core.js/3.6.0 Node.js/16.14.0 (win32; x64)',
      authorization: 'token [REDACTED]'
    },
    request: { hook: [Function: bound bound register] }
  }
}
```

Any help would be greatly appreciated to figure out why octokit returns 404 despite that I am able to log in from my browser.

Thanks,
Abhi

**Comment by timrogers** [comment, marked as answer]
Hi @abhimehta23! Thanks for reaching out.

From the logs you've shared, it looks like your request is going to `https://github.com/{owner}/{repo}/pulls?branch=abhi` - i.e. `{owner}` and `{repo}` are not getting replaced with the right values.

Can you share your code so we can help you debug?

**Reply by abhimehta23** [reply]
Hi @timrogers I changed that to redact my exact link. Here is the link it resolves to and when I copy and paste that link on the browser it does take me to the pull request since I am already signed into the browser with my creds. See original link for reference: `https://github.com/Arris-Composites/RecipeJson/pulls?branch=abhi`

**Reply by abhimehta23** [reply]
Hi @timrogers see code below for reference:

```
const { Octokit } = require ('octokit');

const octokit = new Octokit({
    auth: Process.env.GIT_Token,
    baseUrl: 'https://github.com'
})

const getBranch = async() => {
    let branchNames = await octokit.request('GET /{owner}/{repo}/pulls', {
        owner: 'Arris-Composites',
        repo: 'RecipeJson',
        branch: 'abhi'
    })
}
```

**Reply by timrogers** [reply]
Okay! I understand. You should be hitting `api.github.com` for the API, not just `github.com`, so I think something might be wrong with your Octokit config. How are you configuring it?

**Reply by timrogers** [reply]
Ah, thanks for the code snippet! If you’re using GitHub.com, you don’t need to set a `baseUrl`. We’ll use the right one by default so your request will hit the API rather than the web UI.

**Reply by gr2m** [reply]
It's `GET /repos/{owner}/{repo}/pulls`. And the `baseUrl` is incorrect, should be `https://api.github.com` and for GHES something like `https://github.acme-inc.com/api/v3`

**Reply by abhimehta23** [reply]
@gr2m @timrogers baseUrl with https://api.github.com also did not work with link (https://api.github.com/Arris-Composites/RecipeJson/pulls?branch=abhi). Give me 404 again. 
@gr2m should I be changing the baseUrl to `https://github.acme-inc.com/api/v3`?

**Reply by abhimehta23** [reply]
updating the baseUrl and get method now I am getting status: 500.

```
`throw new requestError.RequestError(error.message, 500, {
          ^

RequestError [HttpError]: request to https://github.acme-inc.com/api/v3/repos/Arris-Composites/RecipeJson/pulls?branch=abhi failed, reason: connect ECONNREFUSED 46.23.69.44:443
    at C:\Users\abhinav\Documents\Projects\OctokitCLI\node_modules\@octokit\request\dist-node\index.js:108:11
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Job.doExecute (C:\Users\abhinav\Documents\Projects\OctokitCLI\node_modules\bottleneck\light.js:405:18) {
  status: 500,
  request: {
    method: 'GET',
    url: 'https://github.acme-inc.com/api/v3/repos/Arris-Composites/RecipeJson/pulls?branch=abhi',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'octokit.js/1.8.1 octokit-core.js/3.6.0 Node.js/16.14.0 (win32; x64)',
      authorization: 'token [REDACTED]'
    },
    request: {
      hook: [Function: bound bound register],
      retryCount: 3,
      retries: 3,
      retryAfter: 16
    }
  }
}`
```

**Reply by timrogers** [reply]
@abhimehta23 When you’re viewing your PRs, are you on GitHub.com? Or are you on another website?

If you’re on GitHub.com, can you remove the `baseUrl` option completely and try the following code?:

```

octokit.request('GET /repos/{owner}/{repo}/pulls', {
        owner: 'Arris-Composites',
        repo: 'RecipeJson',
        branch: 'abhi'
    })

```

**Reply by abhimehta23** [reply]
@gr2m @timrogers Now I am getting status 200 and listing all pull requests. I changed base url to baseUrl: 'https://api.github.com' and Get request to `'GET /repos/{owner}/{repo}/pulls'`

**Reply by abhimehta23** [reply]
@timrogers removed the baseUrl and I am still getting pull requests. Thank you so much!

**Reply by timrogers** [reply]
Great! ✨ Glad we could help!

**Comment by abhimehta23** [comment]
@timrogers quick question - are these API's only applicable for Github or they can be used with GitLab?

**Reply by timrogers** [reply]
GitHub only! I’m sure that GitLab have their own API though.
